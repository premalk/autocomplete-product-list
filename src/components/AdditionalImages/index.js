import { array, func, bool } from 'prop-types';
import Modal from 'react-bootstrap/Modal';

// eslint-disable-next-line react/prop-types
const AdditionalImages = ({ isOpen, setIsOpen, imagelist }) => {
  return (
    <Modal size="sm" show={isOpen} onHide={() => setIsOpen(false)}>
      <Modal.Header closeButton>
        <Modal.Title>Additional images</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        {imagelist.length &&
          imagelist.map((val, index) => (
            <div key={index}>
              <img className="modal-images" src={val} />
            </div>
          ))}
      </Modal.Body>
    </Modal>
  );
};

AdditionalImages.propTypes = {
  isOpen: bool,
  setIsOpen: func,
  imagelist: array
};

export default AdditionalImages;
