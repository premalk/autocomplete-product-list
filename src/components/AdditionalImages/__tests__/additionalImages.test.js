import { render } from '@testing-library/react';
import AdditionalImages from '../';

const mockData = {
  isOpen: true,
  setIsOpen: jest.fn(),
  imagelist: [
    'https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg',
    'https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg',
    'https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@18.jpg'
  ]
};

describe('Testing Modal', () => {
  const modal = render(
    <AdditionalImages
      isOpen={mockData.isOpen}
      setIsOpen={mockData.setIsOpen}
      imagelist={mockData.imagelist}
    />
  );

  test('Snapshot Test of modal', () => {
    expect(modal).toMatchSnapshot();
  });
});
