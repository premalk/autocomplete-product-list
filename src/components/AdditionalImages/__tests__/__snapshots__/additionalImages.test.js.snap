// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Modal Snapshot Test of modal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-sm"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              Additional images
            </div>
            <button
              aria-label="Close"
              class="btn-close"
              type="button"
            />
          </div>
          <div
            class="modal-body"
          >
            <div>
              <img
                class="modal-images"
                src="https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg"
              />
            </div>
            <div>
              <img
                class="modal-images"
                src="https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg"
              />
            </div>
            <div>
              <img
                class="modal-images"
                src="https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@18.jpg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
